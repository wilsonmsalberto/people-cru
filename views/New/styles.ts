import styled from '@emotion/styled';

import { Button } from 'ui/components/Button';

type StyledComponentProps = {
  theme: Record<any, string>;
};

export const Wrapper = styled.section`
  margin-top: 4rem;
  border-radius: 1rem;
  overflow: hidden;
  box-shadow: 0.6rem 0.6rem 5.4rem rgba(0, 0, 0, 0.05);
  background-color: ${(props: StyledComponentProps): string => props.theme.tableRow};
`;

export const TitleWrapper = styled.div`
  padding: 3.6rem;
  border-bottom: 1px solid ${(props: StyledComponentProps): string => props.theme.separator};
`;

export const Title = styled.h1`
  font-weight: 500;
  font-size: 2.2rem;
  line-height: 2.7rem;
`;

export const SubTitle = styled.span`
  font-size: 1.4rem;
  line-height: 2.2rem;
  color: ${(props: StyledComponentProps): string => props.theme.default};
`;

export const FormInnerWrapper = styled.fieldset`
  padding: 4.8rem 24rem;
`;

export const FormFooter = styled.fieldset`
  padding: 3.6rem 0;
  text-align: center;
  background-color: ${(props: StyledComponentProps): string => props.theme.tableFooter};
`;

export const SubmitButton = styled(Button)`
  width: 14.5rem;
  margin: 0 0.8rem;
  box-shadow: none;
`;
