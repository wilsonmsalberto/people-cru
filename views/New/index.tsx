import React from 'react';

import PersonForm from 'ui/blocks/PersonForm';

import { SubTitle, Title, TitleWrapper, Wrapper } from './styles';

const NewPerson = (): React.ReactElement => {
  return (
    <>
      <Wrapper>
        <TitleWrapper>
          <Title>Add a new employee</Title>
          <SubTitle>Fill out the information of your new employee</SubTitle>
        </TitleWrapper>

        <PersonForm submitText="Add employee" />
      </Wrapper>
    </>
  );
};

export default NewPerson;
