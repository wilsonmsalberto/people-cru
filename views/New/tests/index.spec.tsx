import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { matchers } from 'jest-emotion';

import New from '..';

expect.extend(matchers);

describe('New Page', () => {
  describe('Rendering', () => {
    it('should render properly', () => {
      const { asFragment } = render(<New />);

      expect(asFragment()).toMatchSnapshot();
    });
  });
});
