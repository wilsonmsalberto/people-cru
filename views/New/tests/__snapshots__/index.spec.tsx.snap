// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New Page Rendering should render properly 1`] = `
<DocumentFragment>
  <section
    class="css-p7cj8y"
  >
    <div
      class="css-2lunc9"
    >
      <h1
        class="css-1g2rnju"
      >
        Add a new employee
      </h1>
      <span
        class="css-1tvovr2"
      >
        Fill out the information of your new employee
      </span>
    </div>
    <form>
      <fieldset
        class="css-v13m9r"
      >
        <div
          class="css-fsck6u"
        >
          <label
            for="name"
          >
            Name
          </label>
          <input
            id="name"
            placeholder="e.g. Jane Doe"
            type="text"
            value=""
          />
          <span>
            First and last names
          </span>
        </div>
        <div
          class="css-fsck6u"
        >
          <label
            for="birthDate"
          >
            Birthdate
          </label>
          <input
            id="birthDate"
            placeholder="e.g. 17/02/1990"
            type="text"
            value=""
          />
          <span>
            DD/MM/YYYY
          </span>
        </div>
        <div
          class="css-fsck6u"
        >
          <label
            for="job"
          >
            Job title
          </label>
          <input
            id="job"
            placeholder="e.g. Product manager"
            type="text"
            value=""
          />
          <span>
            What is their role?
          </span>
        </div>
        <div
          class="css-1383itu"
        >
          <label
            for="country"
          >
            Country
          </label>
          <div
            class="css-5zrwfr"
          >
            <select
              id="country"
            >
              <option
                disabled=""
                hidden=""
                value="DEFAULT"
              >
                Select their Country
              </option>
              <option
                value="Afghanistan"
              >
                Afghanistan
              </option>
              <option
                value="Albania"
              >
                Albania
              </option>
              <option
                value="Algeria"
              >
                Algeria
              </option>
              <option
                value="American Samoa"
              >
                American Samoa
              </option>
              <option
                value="Andorra"
              >
                Andorra
              </option>
              <option
                value="Angola"
              >
                Angola
              </option>
              <option
                value="Anguilla"
              >
                Anguilla
              </option>
              <option
                value="Antarctica"
              >
                Antarctica
              </option>
              <option
                value="Antigua and Barbuda"
              >
                Antigua and Barbuda
              </option>
              <option
                value="Argentina"
              >
                Argentina
              </option>
              <option
                value="Armenia"
              >
                Armenia
              </option>
              <option
                value="Aruba"
              >
                Aruba
              </option>
              <option
                value="Australia"
              >
                Australia
              </option>
              <option
                value="Austria"
              >
                Austria
              </option>
              <option
                value="Azerbaijan"
              >
                Azerbaijan
              </option>
              <option
                value="Bahamas (the)"
              >
                Bahamas (the)
              </option>
              <option
                value="Bahrain"
              >
                Bahrain
              </option>
              <option
                value="Bangladesh"
              >
                Bangladesh
              </option>
              <option
                value="Barbados"
              >
                Barbados
              </option>
              <option
                value="Belarus"
              >
                Belarus
              </option>
              <option
                value="Belgium"
              >
                Belgium
              </option>
              <option
                value="Belize"
              >
                Belize
              </option>
              <option
                value="Benin"
              >
                Benin
              </option>
              <option
                value="Bermuda"
              >
                Bermuda
              </option>
              <option
                value="Bhutan"
              >
                Bhutan
              </option>
              <option
                value="Bolivia (Plurinational State of)"
              >
                Bolivia (Plurinational State of)
              </option>
              <option
                value="Bonaire, Sint Eustatius and Saba"
              >
                Bonaire, Sint Eustatius and Saba
              </option>
              <option
                value="Bosnia and Herzegovina"
              >
                Bosnia and Herzegovina
              </option>
              <option
                value="Botswana"
              >
                Botswana
              </option>
              <option
                value="Bouvet Island"
              >
                Bouvet Island
              </option>
              <option
                value="Brazil"
              >
                Brazil
              </option>
              <option
                value="British Indian Ocean Territory (the)"
              >
                British Indian Ocean Territory (the)
              </option>
              <option
                value="Brunei Darussalam"
              >
                Brunei Darussalam
              </option>
              <option
                value="Bulgaria"
              >
                Bulgaria
              </option>
              <option
                value="Burkina Faso"
              >
                Burkina Faso
              </option>
              <option
                value="Burundi"
              >
                Burundi
              </option>
              <option
                value="Cabo Verde"
              >
                Cabo Verde
              </option>
              <option
                value="Cambodia"
              >
                Cambodia
              </option>
              <option
                value="Cameroon"
              >
                Cameroon
              </option>
              <option
                value="Canada"
              >
                Canada
              </option>
              <option
                value="Cayman Islands (the)"
              >
                Cayman Islands (the)
              </option>
              <option
                value="Central African Republic (the)"
              >
                Central African Republic (the)
              </option>
              <option
                value="Chad"
              >
                Chad
              </option>
              <option
                value="Chile"
              >
                Chile
              </option>
              <option
                value="China"
              >
                China
              </option>
              <option
                value="Christmas Island"
              >
                Christmas Island
              </option>
              <option
                value="Cocos (Keeling) Islands (the)"
              >
                Cocos (Keeling) Islands (the)
              </option>
              <option
                value="Colombia"
              >
                Colombia
              </option>
              <option
                value="Comoros (the)"
              >
                Comoros (the)
              </option>
              <option
                value="Congo (the Democratic Republic of the)"
              >
                Congo (the Democratic Republic of the)
              </option>
              <option
                value="Congo (the)"
              >
                Congo (the)
              </option>
              <option
                value="Cook Islands (the)"
              >
                Cook Islands (the)
              </option>
              <option
                value="Costa Rica"
              >
                Costa Rica
              </option>
              <option
                value="Croatia"
              >
                Croatia
              </option>
              <option
                value="Cuba"
              >
                Cuba
              </option>
              <option
                value="Curaçao"
              >
                Curaçao
              </option>
              <option
                value="Cyprus"
              >
                Cyprus
              </option>
              <option
                value="Czechia"
              >
                Czechia
              </option>
              <option
                value="Côte d'Ivoire"
              >
                Côte d'Ivoire
              </option>
              <option
                value="Denmark"
              >
                Denmark
              </option>
              <option
                value="Djibouti"
              >
                Djibouti
              </option>
              <option
                value="Dominica"
              >
                Dominica
              </option>
              <option
                value="Dominican Republic (the)"
              >
                Dominican Republic (the)
              </option>
              <option
                value="Ecuador"
              >
                Ecuador
              </option>
              <option
                value="Egypt"
              >
                Egypt
              </option>
              <option
                value="El Salvador"
              >
                El Salvador
              </option>
              <option
                value="Equatorial Guinea"
              >
                Equatorial Guinea
              </option>
              <option
                value="Eritrea"
              >
                Eritrea
              </option>
              <option
                value="Estonia"
              >
                Estonia
              </option>
              <option
                value="Eswatini"
              >
                Eswatini
              </option>
              <option
                value="Ethiopia"
              >
                Ethiopia
              </option>
              <option
                value="Falkland Islands (the) [Malvinas]"
              >
                Falkland Islands (the) [Malvinas]
              </option>
              <option
                value="Faroe Islands (the)"
              >
                Faroe Islands (the)
              </option>
              <option
                value="Fiji"
              >
                Fiji
              </option>
              <option
                value="Finland"
              >
                Finland
              </option>
              <option
                value="France"
              >
                France
              </option>
              <option
                value="French Guiana"
              >
                French Guiana
              </option>
              <option
                value="French Polynesia"
              >
                French Polynesia
              </option>
              <option
                value="French Southern Territories (the)"
              >
                French Southern Territories (the)
              </option>
              <option
                value="Gabon"
              >
                Gabon
              </option>
              <option
                value="Gambia (the)"
              >
                Gambia (the)
              </option>
              <option
                value="Georgia"
              >
                Georgia
              </option>
              <option
                value="Germany"
              >
                Germany
              </option>
              <option
                value="Ghana"
              >
                Ghana
              </option>
              <option
                value="Gibraltar"
              >
                Gibraltar
              </option>
              <option
                value="Greece"
              >
                Greece
              </option>
              <option
                value="Greenland"
              >
                Greenland
              </option>
              <option
                value="Grenada"
              >
                Grenada
              </option>
              <option
                value="Guadeloupe"
              >
                Guadeloupe
              </option>
              <option
                value="Guam"
              >
                Guam
              </option>
              <option
                value="Guatemala"
              >
                Guatemala
              </option>
              <option
                value="Guernsey"
              >
                Guernsey
              </option>
              <option
                value="Guinea"
              >
                Guinea
              </option>
              <option
                value="Guinea-Bissau"
              >
                Guinea-Bissau
              </option>
              <option
                value="Guyana"
              >
                Guyana
              </option>
              <option
                value="Haiti"
              >
                Haiti
              </option>
              <option
                value="Heard Island and McDonald Islands"
              >
                Heard Island and McDonald Islands
              </option>
              <option
                value="Holy See (the)"
              >
                Holy See (the)
              </option>
              <option
                value="Honduras"
              >
                Honduras
              </option>
              <option
                value="Hong Kong"
              >
                Hong Kong
              </option>
              <option
                value="Hungary"
              >
                Hungary
              </option>
              <option
                value="Iceland"
              >
                Iceland
              </option>
              <option
                value="India"
              >
                India
              </option>
              <option
                value="Indonesia"
              >
                Indonesia
              </option>
              <option
                value="Iran (Islamic Republic of)"
              >
                Iran (Islamic Republic of)
              </option>
              <option
                value="Iraq"
              >
                Iraq
              </option>
              <option
                value="Ireland"
              >
                Ireland
              </option>
              <option
                value="Isle of Man"
              >
                Isle of Man
              </option>
              <option
                value="Israel"
              >
                Israel
              </option>
              <option
                value="Italy"
              >
                Italy
              </option>
              <option
                value="Jamaica"
              >
                Jamaica
              </option>
              <option
                value="Japan"
              >
                Japan
              </option>
              <option
                value="Jersey"
              >
                Jersey
              </option>
              <option
                value="Jordan"
              >
                Jordan
              </option>
              <option
                value="Kazakhstan"
              >
                Kazakhstan
              </option>
              <option
                value="Kenya"
              >
                Kenya
              </option>
              <option
                value="Kiribati"
              >
                Kiribati
              </option>
              <option
                value="Korea (the Democratic People's Republic of)"
              >
                Korea (the Democratic People's Republic of)
              </option>
              <option
                value="Korea (the Republic of)"
              >
                Korea (the Republic of)
              </option>
              <option
                value="Kuwait"
              >
                Kuwait
              </option>
              <option
                value="Kyrgyzstan"
              >
                Kyrgyzstan
              </option>
              <option
                value="Lao People's Democratic Republic (the)"
              >
                Lao People's Democratic Republic (the)
              </option>
              <option
                value="Latvia"
              >
                Latvia
              </option>
              <option
                value="Lebanon"
              >
                Lebanon
              </option>
              <option
                value="Lesotho"
              >
                Lesotho
              </option>
              <option
                value="Liberia"
              >
                Liberia
              </option>
              <option
                value="Libya"
              >
                Libya
              </option>
              <option
                value="Liechtenstein"
              >
                Liechtenstein
              </option>
              <option
                value="Lithuania"
              >
                Lithuania
              </option>
              <option
                value="Luxembourg"
              >
                Luxembourg
              </option>
              <option
                value="Macao"
              >
                Macao
              </option>
              <option
                value="Madagascar"
              >
                Madagascar
              </option>
              <option
                value="Malawi"
              >
                Malawi
              </option>
              <option
                value="Malaysia"
              >
                Malaysia
              </option>
              <option
                value="Maldives"
              >
                Maldives
              </option>
              <option
                value="Mali"
              >
                Mali
              </option>
              <option
                value="Malta"
              >
                Malta
              </option>
              <option
                value="Marshall Islands (the)"
              >
                Marshall Islands (the)
              </option>
              <option
                value="Martinique"
              >
                Martinique
              </option>
              <option
                value="Mauritania"
              >
                Mauritania
              </option>
              <option
                value="Mauritius"
              >
                Mauritius
              </option>
              <option
                value="Mayotte"
              >
                Mayotte
              </option>
              <option
                value="Mexico"
              >
                Mexico
              </option>
              <option
                value="Micronesia (Federated States of)"
              >
                Micronesia (Federated States of)
              </option>
              <option
                value="Moldova (the Republic of)"
              >
                Moldova (the Republic of)
              </option>
              <option
                value="Monaco"
              >
                Monaco
              </option>
              <option
                value="Mongolia"
              >
                Mongolia
              </option>
              <option
                value="Montenegro"
              >
                Montenegro
              </option>
              <option
                value="Montserrat"
              >
                Montserrat
              </option>
              <option
                value="Morocco"
              >
                Morocco
              </option>
              <option
                value="Mozambique"
              >
                Mozambique
              </option>
              <option
                value="Myanmar"
              >
                Myanmar
              </option>
              <option
                value="Namibia"
              >
                Namibia
              </option>
              <option
                value="Nauru"
              >
                Nauru
              </option>
              <option
                value="Nepal"
              >
                Nepal
              </option>
              <option
                value="Netherlands (the)"
              >
                Netherlands (the)
              </option>
              <option
                value="New Caledonia"
              >
                New Caledonia
              </option>
              <option
                value="New Zealand"
              >
                New Zealand
              </option>
              <option
                value="Nicaragua"
              >
                Nicaragua
              </option>
              <option
                value="Niger (the)"
              >
                Niger (the)
              </option>
              <option
                value="Nigeria"
              >
                Nigeria
              </option>
              <option
                value="Niue"
              >
                Niue
              </option>
              <option
                value="Norfolk Island"
              >
                Norfolk Island
              </option>
              <option
                value="Northern Mariana Islands (the)"
              >
                Northern Mariana Islands (the)
              </option>
              <option
                value="Norway"
              >
                Norway
              </option>
              <option
                value="Oman"
              >
                Oman
              </option>
              <option
                value="Pakistan"
              >
                Pakistan
              </option>
              <option
                value="Palau"
              >
                Palau
              </option>
              <option
                value="Palestine, State of"
              >
                Palestine, State of
              </option>
              <option
                value="Panama"
              >
                Panama
              </option>
              <option
                value="Papua New Guinea"
              >
                Papua New Guinea
              </option>
              <option
                value="Paraguay"
              >
                Paraguay
              </option>
              <option
                value="Peru"
              >
                Peru
              </option>
              <option
                value="Philippines (the)"
              >
                Philippines (the)
              </option>
              <option
                value="Pitcairn"
              >
                Pitcairn
              </option>
              <option
                value="Poland"
              >
                Poland
              </option>
              <option
                value="Portugal"
              >
                Portugal
              </option>
              <option
                value="Puerto Rico"
              >
                Puerto Rico
              </option>
              <option
                value="Qatar"
              >
                Qatar
              </option>
              <option
                value="Republic of North Macedonia"
              >
                Republic of North Macedonia
              </option>
              <option
                value="Romania"
              >
                Romania
              </option>
              <option
                value="Russian Federation (the)"
              >
                Russian Federation (the)
              </option>
              <option
                value="Rwanda"
              >
                Rwanda
              </option>
              <option
                value="Réunion"
              >
                Réunion
              </option>
              <option
                value="Saint Barthélemy"
              >
                Saint Barthélemy
              </option>
              <option
                value="Saint Helena, Ascension and Tristan da Cunha"
              >
                Saint Helena, Ascension and Tristan da Cunha
              </option>
              <option
                value="Saint Kitts and Nevis"
              >
                Saint Kitts and Nevis
              </option>
              <option
                value="Saint Lucia"
              >
                Saint Lucia
              </option>
              <option
                value="Saint Martin (French part)"
              >
                Saint Martin (French part)
              </option>
              <option
                value="Saint Pierre and Miquelon"
              >
                Saint Pierre and Miquelon
              </option>
              <option
                value="Saint Vincent and the Grenadines"
              >
                Saint Vincent and the Grenadines
              </option>
              <option
                value="Samoa"
              >
                Samoa
              </option>
              <option
                value="San Marino"
              >
                San Marino
              </option>
              <option
                value="Sao Tome and Principe"
              >
                Sao Tome and Principe
              </option>
              <option
                value="Saudi Arabia"
              >
                Saudi Arabia
              </option>
              <option
                value="Senegal"
              >
                Senegal
              </option>
              <option
                value="Serbia"
              >
                Serbia
              </option>
              <option
                value="Seychelles"
              >
                Seychelles
              </option>
              <option
                value="Sierra Leone"
              >
                Sierra Leone
              </option>
              <option
                value="Singapore"
              >
                Singapore
              </option>
              <option
                value="Sint Maarten (Dutch part)"
              >
                Sint Maarten (Dutch part)
              </option>
              <option
                value="Slovakia"
              >
                Slovakia
              </option>
              <option
                value="Slovenia"
              >
                Slovenia
              </option>
              <option
                value="Solomon Islands"
              >
                Solomon Islands
              </option>
              <option
                value="Somalia"
              >
                Somalia
              </option>
              <option
                value="South Africa"
              >
                South Africa
              </option>
              <option
                value="South Georgia and the South Sandwich Islands"
              >
                South Georgia and the South Sandwich Islands
              </option>
              <option
                value="South Sudan"
              >
                South Sudan
              </option>
              <option
                value="Spain"
              >
                Spain
              </option>
              <option
                value="Sri Lanka"
              >
                Sri Lanka
              </option>
              <option
                value="Sudan (the)"
              >
                Sudan (the)
              </option>
              <option
                value="Suriname"
              >
                Suriname
              </option>
              <option
                value="Svalbard and Jan Mayen"
              >
                Svalbard and Jan Mayen
              </option>
              <option
                value="Sweden"
              >
                Sweden
              </option>
              <option
                value="Switzerland"
              >
                Switzerland
              </option>
              <option
                value="Syrian Arab Republic"
              >
                Syrian Arab Republic
              </option>
              <option
                value="Taiwan (Province of China)"
              >
                Taiwan (Province of China)
              </option>
              <option
                value="Tajikistan"
              >
                Tajikistan
              </option>
              <option
                value="Tanzania, United Republic of"
              >
                Tanzania, United Republic of
              </option>
              <option
                value="Thailand"
              >
                Thailand
              </option>
              <option
                value="Timor-Leste"
              >
                Timor-Leste
              </option>
              <option
                value="Togo"
              >
                Togo
              </option>
              <option
                value="Tokelau"
              >
                Tokelau
              </option>
              <option
                value="Tonga"
              >
                Tonga
              </option>
              <option
                value="Trinidad and Tobago"
              >
                Trinidad and Tobago
              </option>
              <option
                value="Tunisia"
              >
                Tunisia
              </option>
              <option
                value="Turkey"
              >
                Turkey
              </option>
              <option
                value="Turkmenistan"
              >
                Turkmenistan
              </option>
              <option
                value="Turks and Caicos Islands (the)"
              >
                Turks and Caicos Islands (the)
              </option>
              <option
                value="Tuvalu"
              >
                Tuvalu
              </option>
              <option
                value="Uganda"
              >
                Uganda
              </option>
              <option
                value="Ukraine"
              >
                Ukraine
              </option>
              <option
                value="United Arab Emirates (the)"
              >
                United Arab Emirates (the)
              </option>
              <option
                value="United Kingdom of Great Britain and Northern Ireland (the)"
              >
                United Kingdom of Great Britain and Northern Ireland (the)
              </option>
              <option
                value="United States Minor Outlying Islands (the)"
              >
                United States Minor Outlying Islands (the)
              </option>
              <option
                value="United States of America (the)"
              >
                United States of America (the)
              </option>
              <option
                value="Uruguay"
              >
                Uruguay
              </option>
              <option
                value="Uzbekistan"
              >
                Uzbekistan
              </option>
              <option
                value="Vanuatu"
              >
                Vanuatu
              </option>
              <option
                value="Venezuela (Bolivarian Republic of)"
              >
                Venezuela (Bolivarian Republic of)
              </option>
              <option
                value="Viet Nam"
              >
                Viet Nam
              </option>
              <option
                value="Virgin Islands (British)"
              >
                Virgin Islands (British)
              </option>
              <option
                value="Virgin Islands (U.S.)"
              >
                Virgin Islands (U.S.)
              </option>
              <option
                value="Wallis and Futuna"
              >
                Wallis and Futuna
              </option>
              <option
                value="Western Sahara"
              >
                Western Sahara
              </option>
              <option
                value="Yemen"
              >
                Yemen
              </option>
              <option
                value="Zambia"
              >
                Zambia
              </option>
              <option
                value="Zimbabwe"
              >
                Zimbabwe
              </option>
              <option
                value="Åland Islands"
              >
                Åland Islands
              </option>
            </select>
          </div>
          <span>
            Where are they based?
          </span>
        </div>
        <div
          class="css-fsck6u"
        >
          <label
            for="salary"
          >
            Salary
          </label>
          <input
            id="salary"
            placeholder="e.g. 5000"
            type="text"
            value=""
          />
          <span>
            Gross monthly salary
          </span>
        </div>
      </fieldset>
      <fieldset
        class="css-1w1h6zs"
      >
        <button
          class="css-1uoc2nl"
          type="button"
        >
          Cancel
        </button>
        <button
          class="css-1uoc2nl"
          type="submit"
        >
          Add employee
        </button>
      </fieldset>
    </form>
  </section>
</DocumentFragment>
`;
